### PR TITLE
Feature usage update

### PR DESCRIPTION
**Pull request by kjgarza**
Managing these changes from here. Not ready to merge at this point

### CODE REVIEW

**Comment by kjgarza** [comment]
@mfenner I commented the test for the new features because I didn't use stubbing (I tested against an actual queue) so I will need to change that in https://github.com/datacite/toccatore/issues/4 but I will like to merge to check the whole workflow.

File: Gemfile.lock [hunk 1]
```diff
@@ -3,31 +3,42 @@ PATH
...
       namae (~> 0.11.0)
-      slack-notifier (~> 2.1)
+      slack-notifier (= 2.2.2)
       thor (~> 0.19)
 
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (4.2.9)
+    activesupport (4.2.10)
```
Comment (mfenner) [inline Gemfile.lock]: Consider upgrading to activesupport 5.x.

File: lib/toccatore/usage_update.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        error_total = 0
+        Array(items).each do |item|
+          error_total += push_item(item, options)
+        end
+        error_total
+      end
+    end
+
+    def metrics_url
+      "https://metrics.test.datacite.org/reports/"
```
Comment (mfenner) [inline lib/toccatore/usage_update.rb]: Are there plans to change the subdomain?

File: lib/toccatore/usage_update.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    end
+
+    def source_id
+      "usage_update"
+    end
+
+    def query
+      "relatedIdentifier:DOI\\:*"
```
Comment (mfenner) [inline lib/toccatore/usage_update.rb]: This looks like a Solr query. Is this intended?

File: lib/toccatore/usage_update.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      created = header.fetch("created")
+      Array.wrap(items).reduce([]) do |x, item|
+        data = {}
+        data[:doi] = item.dig("dataset-id").first.dig("value")
+        data[:pid] = normalize_doi(data[:doi])
+        data[:created] = created
+        data[:report_id] = report_id
+        data[:created_at] = created
+
+        instances = item.dig("performance").first.dig("instance")
```
Comment (mfenner) [inline lib/toccatore/usage_update.rb]: You can simplify to `item.dig("performance", 0, "instance")`.

File: lib/toccatore/usage_update.rb [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+        return x += [OpenStruct.new(body: { "errors" => "There are too many instances. There can only be 4" })] if instances.size > 8
+     
+        x += Array.wrap(instances).reduce([]) do |ssum, instance|
+          data[:count] = instance.dig("count")
+          event_type = "#{instance.dig("metric-type")}-#{instance.dig("access-method")}"
+          ssum << format_event(event_type, data, options)
```
Comment (mfenner) [inline lib/toccatore/usage_update.rb]: You probably need a new line with `ssum`.

**Comment by mfenner** [comment]
What is not clear to me is how this will be run: triggered every 24h via lambda like the other toccatore methods, or as a constantly running service. The SQS queue is a better fit for the latter.

File: lib/toccatore/usage_update.rb (hunk 2 above)
Comment (kjgarza) [reply lib/toccatore/usage_update.rb]: fixed 9c4c335

File: lib/toccatore/usage_update.rb (hunk 3 above)
Comment (kjgarza) [reply lib/toccatore/usage_update.rb]: fixed 9c4c335

File: lib/toccatore/usage_update.rb (hunk 4 above)
Comment (kjgarza) [reply lib/toccatore/usage_update.rb]: fixed 9c4c335

File: lib/toccatore/usage_update.rb (hunk 5 above)
Comment (kjgarza) [reply lib/toccatore/usage_update.rb]: fixed 9c4c335

**Pull request merged**